### PR TITLE
Partner reporting dates 

### PR DESCRIPTION
**Pull request by daniel-tabarcea**
- partner reporting requirement dates now displayed in the interventions details page
- connects unicef/etools-issues#832

### CODE REVIEW

File: pmp/app-elements/interventions/reports/intervention-reporting-requirements.html [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+<link rel="import" href="../../../../bower_components/etools-dialog/dynamic-dialog-behavior.html">
+<link rel="import" href="../../../../bower_components/etools-ajax/etools-ajax-request-behavior.html">
 
+<link rel="import" href="../../../app-config/app-endpoints-behavior.html">
+<link rel="import" href="../../../app-behaviors/ajax-server-errors-behavior.html">
 <link rel="import" href="../../../../styles/grid-layout-styles.html">
 <link rel="import" href="../../../../styles/shared-styles.html">
-
+<link rel="import" href="../intervention-data/reporting-requirements-data.html">
+<link rel="import" href="../intervention-data/reporting-requirement-item-data.html">
```
Comment (acory) [inline pmp/app-elements/interventions/reports/intervention-reporting-requirements.html]: These files are not in the solution
/intervention-data/reporting-requirements-data.html
/intervention-data/reporting-requirement-item-data.html

File: pmp/app-elements/interventions/reports/intervention-reporting-requirements.html [hunk 2]
```diff
@@ -149,10 +88,215 @@
...
+          'requiremenet-delete': '_handleRequirementDelete'
+        },
+
+        ready: function() {
+          this._createRequirementsDialog();
+          this._createDeleteConfirmationDialog();
+        },
+
+        detached: function() {
+          if (this.requirementsDialog) {
```
Comment (adi130987) [inline pmp/app-elements/interventions/reports/intervention-reporting-requirements.html]: I think you can use `removeDialog` as well

File: pmp/app-elements/interventions/reports/intervention-reporting-requirements.html [hunk 3]
```diff
@@ -149,10 +88,215 @@
...
+
+        _interventionIdChanged: function(newId) {
+          let self = this;
+
+          if (newId) {
+            let endpoint = this.getEndpoint(this.endpoints.LIST, {intervId: newId});
+            this.sendRequest({method:'GET', endpoint: endpoint})
+              .then(function(response) {
+                self.set('reportingRequirements', response);
+              });;
```
Comment (adi130987) [inline pmp/app-elements/interventions/reports/intervention-reporting-requirements.html]: double `;`

File: pmp/app-elements/common-elements/etools-date-input.html [hunk 4]
```diff
@@ -149,6 +155,13 @@
           this.set('invalid', !valid);
           return valid;
         },
+
+        makeValid: function() {
```
Comment (adi130987) [inline pmp/app-elements/common-elements/etools-date-input.html]: `makeValid` should be renamed to something else. You don't make the field value, you're just resetting the invalid state.

File: pmp/app-elements/interventions/reports/reporting-requirements.html [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+          });
+
+          if (!newReq.id) {
+            this.addRequirement(newReq)
+              .then(function() {
+                self.fire('global-loading', {
+                  active: false,
+                  loadingSource: self.ajaxLoadingMsgSource
+                });
+                self.fire('toast', {text: 'Reporting requirement successfully added', showCloseBtn: true});
```
Comment (acory) [inline pmp/app-elements/interventions/reports/reporting-requirements.html]: Is this message necessary? we don't have it in other similar situations

**Pull request merged**